tests/test_ratings.py

```diff
@@ -164,3 +166,112 @@ def test_delete_rating(selenium, base_url, variables):
+    extension = variables['detail_extension_slug']
+    selenium.get(f'{base_url}/addon/{extension}')
+    addon = Detail(selenium, base_url).wait_for_page_to_load()
+    reviews = addon.ratings.click_all_reviews_link()
+    assert reviews.addon_summary_card.is_displayed()
+    for review in reviews.review_items:
+        assert review.rating_stars.is_displayed()
+        assert review.rating_user.is_displayed()
+        assert review.posting_date.is_displayed()
+
+
+@pytest.mark.nondestructive
+def test_filter_reviews_by_score(selenium, base_url, variables):
+    extension = variables['all_scores_addon']
+    selenium.get(f'{base_url}/addon/{extension}')
+    addon = Detail(selenium, base_url).wait_for_page_to_load()
+    reviews = addon.ratings.click_all_reviews_link()
+    select = Select(reviews.filter_by_score)
+    count = 1
+    # selecting rating score filters from 1 to 5, verifying that the reviews card
+    # header counts reflect the number of reviews available for that rating score
+    # and that the ratings stars for each review reflect the selected score
+    while count < 6:
+        select.select_by_value(str(count))
+        # waiting for the reviews page to be refreshed
+        reviews.wait_for_page_to_load()
+        assert reviews.reviews_title_count == len(reviews.reviews_list)
+        for stars in reviews.review_items:
+            assert len(stars.selected_star) == count
+        count += 1
+
+
+@pytest.mark.nondestructive
+def test_filter_reviews_from_rating_bars(selenium, base_url, variables):
+    extension = variables['all_scores_addon']
+    selenium.get(f'{base_url}/addon/{extension}')
+    addon = Detail(selenium, base_url).wait_for_page_to_load()
+    reviews = addon.ratings.click_all_reviews_link()
+    count = 0
+    # sort reviews based on score by clicking on the ratings bars present
+    # in the AddonSummaryCard; verify that the correct score is displayed
+    # in the reviews card header and in the rating stars associated to each review
+    while count < 5:
+        reviews.score_bars[count].click()
+        # waiting for the reviews page to be refreshed
+        reviews.wait_for_page_to_load()
+        assert reviews.reviews_title_count == len(reviews.reviews_list)
+        for stars in reviews.review_items:
+            assert len(stars.selected_star) == int(reviews.bar_rating_score[count].text)
+        count += 1
+
+
+@pytest.mark.nondestructive
+def test_flag_review_action(selenium, base_url, variables):
+    extension = variables['all_scores_addon']
+    selenium.get(f'{base_url}/addon/{extension}')
+    addon = Detail(selenium, base_url).wait_for_page_to_load()
+    addon.login('regular_user')
+    reviews = addon.ratings.click_all_reviews_link()
+    flag = reviews.review_items
+    # the 'Flag' menu is displayed only for reviews with text
+    # iterating through the list of reviews until a review with text is found
+    count = 0
+    while True:
```

Comment:
Is this supposed to be ```while count``` ?

Reply:
@jrbenny35 If I'm using `while count` the code inside the while loop will not execute (it is not reached). With `while True` the code is doing what is supposed to, i.e. iterate through the list of reviews until the if condition is satisfied (i.e. a rating with text is found).

Reply:
Any way we could use the amount of reviews in the review list as a limit?

Reply:
@jrbenny35 yes, this is a better solution - https://github.com/mozilla/addons-release-tests/pull/260/commits/f45751c475198741191587677362521105341440